src/windows/VibrationProxy.js

```diff
@@ -60,12 +61,43 @@ if (VibrationDevice) {
                 fail(e);
             }
         }, 
-        vibrateWithPattern: function(success, fail, args) {
-            // TODO: Implement with setTimeout.
-            fail('"vibrateWithPattern" is unsupported by this platform.');
+        vibrateWithPattern: function (success, fail, args) {
+            // Cancel current vibrations first
+            module.exports.cancelVibration(function () {
+                // Removes the first zero
+                if (args[0] && args[0][0] === 0) {
+                    args[0].shift(0);
+                }
+
+                patternChainPromise = args[0].reduce(function (previousValue, currentValue, index) {
+                    if (index % 2 === 0) {
+                        return previousValue.then(function () {
+                            return new WinJS.Promise(function (complete, error) {
+                                module.exports.vibrate(function () {}, function(err) {
+                                    console.error(err);
+                                    error(err);
+                                }, [currentValue]);
```

Comment:
We make a lot of references to `currentValue`.  Can we just parse it once and save the integer form?  Maybe we can opportunistically parse all of them on input and if any parse return NaN then we fail fast.